pandas/core/reshape/merge.py

```diff
@@ -1276,6 +1276,11 @@ def _get_join_indexers(left_keys, right_keys, sort=False, how="inner", **kwargs)
         indexers into the left_keys, right_keys
 
     """
+    _how = how
```

Comment:
Hmm this is rather difficult to grok - could you alternately just change `_right_outer_join` within the same module to accept a `sort` keyword? I think that would make things easier

Reply:
the sorting actually happens in `_factorize_keys` [a few lines down](https://github.com/pandas-dev/pandas/pull/27762/files#diff-14fc0a514ad9ce2a73eeaa9ec3fb9406R1292), not downstream in `_right_outer_merge`. So the left and right keys need to be swapped before that call. `_right_outer_join` just implements `_left_outer_join` with left and right swapped; no sorting happens there.

Reply:
I guess what I'm saying is that sorting already happens, but it happens before a distinction is made between left and right joins. So this PR doesn't move where the sorting happens, it just moves where the distinction between left and right joins is made.